etcdctl/ctlv2/command/util.go

```diff
@@ -203,7 +203,7 @@ func mustNewClient(c *cli.Context) client.Client {
 		if debug {
 			fmt.Fprintf(os.Stderr, "start to sync cluster using endpoints(%s)\n", strings.Join(hc.Endpoints(), ","))
 		}
-		ctx, cancel := context.WithTimeout(context.Background(), client.DefaultRequestTimeout)
+		ctx, cancel := contextWithTotalTimeout(c)
```

Comment:
@xiang90 

> also here, we should also add a timeout for the request timeout.

Sorry, can you explain more? This already puts timeout for the command:

```go
func contextWithTotalTimeout(c *cli.Context) (context.Context, context.CancelFunc) {
	return context.WithTimeout(context.Background(), c.GlobalDuration("total-timeout"))
}
``` 

`c.GlobalDuration("timeout")` is already used for `HeaderTimeoutPerRequest` here https://github.com/coreos/etcd/blob/master/etcdctl/ctlv2/command/util.go#L302